Cannot compile with GHC 8.4 or later

If I compile `HGL-3.2.3.1` with GHC 8.4 or later, I get the following error:

```
$ cabal new-install HGL -j1
Resolving dependencies...
Build profile: -w ghc-8.6.2 -O1
In order, the following will be built (use -v for more details):
 - HGL-3.2.3.1 (lib:HGL) (requires build)
Warning: HGL.cabal:10:2: Tabs used as indentation at 10:2, 11:2, 12:2, 13:2,
14:2, 15:2, 16:2, 17:2
Configuring HGL-3.2.3.1...
Preprocessing library for HGL-3.2.3.1..
Building library for HGL-3.2.3.1..
[ 1 of 27] Compiling Graphics.HGL.Internals.Flag ( Graphics/HGL/Internals/Flag.hs, dist/build/Graphics/HGL/Internals/Flag.o )

Graphics/HGL/Internals/Flag.hs:17:1: warning: [-Wtabs]
    Tab character found here, and in three further locations.
    Please use spaces instead.
   |
17 |         ( Flag, newFlag, setFlag, resetFlag
   | ^^^^^^^^
[ 2 of 27] Compiling Graphics.HGL.Internals.Types ( Graphics/HGL/Internals/Types.hs, dist/build/Graphics/HGL/Internals/Types.o )

Graphics/HGL/Internals/Types.hs:63:9: warning: [-Wtabs]
    Tab character found here, and in 26 further locations.
    Please use spaces instead.
   |
63 |   | Dash        -- "-------"
   |         ^^^^^^^^
[ 3 of 27] Compiling Graphics.HGL.Internals.Utilities ( Graphics/HGL/Internals/Utilities.hs, dist/build/Graphics/HGL/Internals/Utilities.o )

Graphics/HGL/Internals/Utilities.hs:17:1: warning: [-Wtabs]
    Tab character found here, and in three further locations.
    Please use spaces instead.
   |
17 |         bracket, bracket_,
   | ^^^^^^^^
[ 4 of 27] Compiling Graphics.HGL.Units ( Graphics/HGL/Units.hs, dist/build/Graphics/HGL/Units.o )

Graphics/HGL/Units.hs:16:1: warning: [-Wtabs]
    Tab character found here, and in four further locations.
    Please use spaces instead.
   |
16 |         ( Point
   | ^^^^^^^^
[ 5 of 27] Compiling Graphics.HGL.X11.Display ( Graphics/HGL/X11/Display.hs, dist/build/Graphics/HGL/X11/Display.o )

Graphics/HGL/X11/Display.hs:3:1: warning: [-Wtabs]
    Tab character found here, and in four further locations.
    Please use spaces instead.
  |
3 |         ( getDisplayName
  | ^^^^^^^^
[ 6 of 27] Compiling Graphics.HGL.X11.Timer ( Graphics/HGL/X11/Timer.hs, dist/build/Graphics/HGL/X11/Timer.o )

Graphics/HGL/X11/Timer.hs:17:1: warning: [-Wtabs]
    Tab character found here, and in two further locations.
    Please use spaces instead.
   |
17 |         ( Timer, new, stop
   | ^^^^^^^^
[ 7 of 27] Compiling Graphics.HGL.X11.Types ( Graphics/HGL/X11/Types.hs, dist/build/Graphics/HGL/X11/Types.o )

Graphics/HGL/X11/Types.hs:17:1: warning: [-Wtabs]
    Tab character found here, and in 7 further locations.
    Please use spaces instead.
   |
17 |         ( DC(..)
   | ^^^^^^^^
[ 8 of 27] Compiling Graphics.HGL.Key ( Graphics/HGL/Key.hs, dist/build/Graphics/HGL/Key.o )

Graphics/HGL/Key.hs:16:1: warning: [-Wtabs]
    Tab character found here, and in 79 further locations.
    Please use spaces instead.
   |
16 |         ( Key                   -- Abstract!
   | ^^^^^^^^
[ 9 of 27] Compiling Graphics.HGL.Internals.Event ( Graphics/HGL/Internals/Event.hs, dist/build/Graphics/HGL/Internals/Event.o )

Graphics/HGL/Internals/Event.hs:17:1: warning: [-Wtabs]
    Tab character found here, and in 39 further locations.
    Please use spaces instead.
   |
17 |         ( Event(..)
   | ^^^^^^^^
[10 of 27] Compiling Graphics.HGL.Internals.Events ( Graphics/HGL/Internals/Events.hs, dist/build/Graphics/HGL/Internals/Events.o )

Graphics/HGL/Internals/Events.hs:23:68: error:
    Module ‘Control.Concurrent.Chan’ does not export ‘isEmptyChan’
   |
23 | import Control.Concurrent.Chan(Chan, newChan, readChan, writeChan, isEmptyChan)
   |                                                                    ^^^^^^^^^^^
Warning: Some package(s) failed to build. Try rerunning with -j1 if you can't
see the error.
```

This is because `isEmptyChan` (which had long been deprecated) was finally removed in GHC 8.4. Is it possible to fix this?

--- a/Graphics/HGL/Internals/Events.hs
+++ b/Graphics/HGL/Internals/Events.hs
@@ -18,9 +18,11 @@ module Graphics.HGL.Internals.Events(
         getTick, sendTick
 	) where
 
+import Control.Concurrent.STM (atomically)
+import Control.Concurrent.STM.TChan(TChan, newTChanIO, readTChan, writeTChan, isEmptyTChan)
+
 import Graphics.HGL.Internals.Event
 import Graphics.HGL.Internals.Flag
-import Control.Concurrent.Chan(Chan, newChan, readChan, writeChan, isEmptyChan)
 
 ----------------------------------------------------------------
 -- Interface
@@ -36,6 +38,8 @@ import Control.Concurrent.Chan(Chan, newChan, readChan, writeChan, isEmptyChan)
 -- myself reimplementing this in Haskell (even in the Win32 version
 -- of the Graphics library).  Exposure events in X11 behave in a
 -- similar way except that they do not overtake other events.)
+
+type Chan = TChan
 
 data Events = Events { events :: Chan Event
                      , tick   :: Flag ()
@@ -62,6 +66,25 @@ isNoEvent evs = isEmptyChan (events evs)
 sendEvent evs = writeChan   (events evs)
 sendTick  evs = setFlag     (tick evs) ()
 getTick   evs = resetFlag   (tick evs)
+
+----------------------------------------------------------------
+-- TChan wrappers
+--
+-- As `isEmptyChan` was removed by
+-- http://hackage.haskell.org/trac/ghc/ticket/4154, we can follow
+-- suggestions in there and just replace Chan with TChan.
+----------------------------------------------------------------
+newChan :: IO (Chan a)
+newChan = newTChanIO
+
+readChan :: Chan a -> IO a
+readChan = atomically . readTChan
+
+writeChan :: Chan a -> a -> IO ()
+writeChan c v = atomically $ writeTChan c v
+
+isEmptyChan :: Chan a -> IO Bool
+isEmptyChan = atomically . isEmptyTChan
 
 ----------------------------------------------------------------
 -- End

--- a/HGL.cabal
+++ b/HGL.cabal
@@ -22,7 +22,7 @@ flag split-base
 
 library
   if flag(split-base)
-    build-depends: base >= 4.4.0.0 && < 5, array
+    build-depends: base >= 4.4.0.0 && < 5, array, stm
   else
     build-depends: base < 2
   exposed-modules:

